src/caUtility.cpp

```diff
@@ -0,0 +1,40 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// This is a simple example of exporting a C++ function to R. You can
+// source this function into an R session using the Rcpp::sourceCpp 
+// function (or via the Source button on the editor toolbar). Learn
+// more about Rcpp at:
+//
+//   http://www.rcpp.org/
+//   http://adv-r.had.co.nz/Rcpp.html
+//   http://gallery.rcpp.org/
+//
+
+// [[Rcpp::export]]
+NumericVector caUtility_rcpp(double aCA, double bCA, double bCAlr) {
+  
+  // allocate variables
+  NumericVector output(11);
+  
+  // angles 
+  NumericVector angles =
+    NumericVector::create(10, 20, 32.5, 50, 72.5) / 90;

```

Comment:
This creates two NumericVectors. More efficient would be to put the divisions inside:
```suggestion
    NumericVector::create(10 / 90., 20 / 90., 32.5 / 90., 50 / 90., 72.5 / 90.);
```
Again: micro-optimization.

Reply:
I also wonder if using `NumericVector` is the best way to go, or whether a `std::vector` or `std::array` would be faster. I suspect `NumericVector` has extra baggage that you may not need.

Reply:
Actually, looking at the R-version, I see now that this vector is an optional argument there. The hardcoded values are just the default values, but in principle you could put any angle array in. Perhaps check whether that indeed will be necessary.